esp-hal-common/src/lib.rs

```diff
@@ -141,23 +140,19 @@ pub mod trapframe {
     pub use xtensa_lx_rt::exception::Context as TrapFrame;
 }
 
+// The `soc` module contains chip-specific implementation details and should not
+// be directly exposed.
+mod soc;
```

Comment:
I think this is also a breaking change as it was available last release: https://docs.rs/esp32c3-hal/0.9.0/esp32c3_hal/. It probably needs an entry in the changelog.

Reply:
also there are things like PSRAM's public API currently - maybe it should be exported elsewhere but definitely a breaking change